mailpoet/lib/Segments/DynamicSegments/FilterDataMapper.php

```diff
@@ -509,6 +510,10 @@ private function createWooCommerce(array $data): DynamicSegmentFilterData {
       $filterData['coupon_code_ids'] = $data['coupon_code_ids'];
       $filterData['days'] = $data['days'];
       $filterData['timeframe'] = $data['timeframe'];
+    } elseif ($data['action'] === WooCommercePurchasedWithAttribute::ACTION) {
+      $filterData['operator'] = $data['operator'];
+      $filterData['attribute_taxonomy_slug'] = $data['attribute_taxonomy_slug'];
+      $filterData['attribute_term_ids'] = $data['attribute_term_ids'];
```

Comment:
Maybe we could add a couple of validations like we have for other filter types ([e.g.](https://github.com/mailpoet/mailpoet/pull/5437/commits/90e35c24325934cf629a34b46df5c8b55211edc2#diff-44dc56dc4354bb109ce8925b08581157b0d1a381f2e399d20204d4ea4fe40455R494-R503)).
This could prevent saving invalid data from the clients. What do you think?

Please note that there is also [a test](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/tests/integration/Segments/DynamicSegments/FilterDataMapperTest.php) for this class so it would be nice to add cases for this filter type.

Reply:
Having it together with a filter definition makes sense, and I agree it is a better place than in the mapper. My concern is that we would skip validation on save and allow saving invalid combinations (maybe due to a change after the JS dependencies update, we may start sending wrong data). But I might be missing something. Are the data validated somehow before saving?

Reply:
@costasovo The validation is running in the mapper as well, currently [here](https://github.com/mailpoet/mailpoet/blob/a870a2f3edec650b70fe3406e5bdcce6bf380c03/mailpoet/lib/Segments/DynamicSegments/FilterDataMapper.php#L518-L518). I don't know of anywhere else we would need to run the validation before saving.

Reply:
Ahh! Perfect! So I missed that. Thank you!